app/controllers/get_involved_controller.rb

```diff
@@ -0,0 +1,94 @@
+class GetInvolvedController < ApplicationController
+  attr_accessor :content_item
+
+  def show
+    load_content_item
+    load_get_involved_data
+    @do_not_show_breadcrumbs = true
+    render template: "content_items/get_involved"
+  end
+
+  def load_content_item
+    content_item = Services.content_store.content_item("/government/get-involved")
+
+    if Services.feature_toggler.use_recommended_related_links?(content_item["links"], request.headers)
+      content_item["links"]["ordered_related_items"] = content_item["links"].fetch("suggested_ordered_related_items", [])
+    end
+
+    @content_item = PresenterBuilder.new(
+      content_item,
+      content_item_path,
+      view_context,
+    ).presenter
+  end
+
+  def load_get_involved_data
+    @open_consultation_count = retrieve_given_document_type("open_consultation")["total"]
+    @closed_consultation_count = retrieve_date_filtered_closed_consultations(12)
+    @next_closing_consultations = [retrieve_next_closing]
+    @recently_opened_consultations = retrieve_new_consultations
+    @recent_consultation_outcomes = retrieve_consultation_outcomes
+    @take_part_pages = sort_take_part(retrieve_given_document_type("take_part")["results"])
+  end
+
+  def retrieve_given_document_type(document_type)
+    Services.publishing_api.get_content_items(document_type: document_type).to_hash
+  end
+
+  def retrieve_special(params)
+    Services.publishing_api.get_content_items(params).to_hash
+  end
+
+  # Aims to find a count of consultations closed in the last n months.
+  # Could feasibly be done via SearchAPI calls but there is no direct or indirect way to search by closing date
```

Comment:
This does concern me a little bit. It's quite unusual for a frontend application to need to talk to the Publishing API (except [for publishing its routes which itself is considered tech debt](https://trello.com/c/hJVUjqbs/197-frontend-apps-publish-content-items-to-publishing-api)) and it definitely feels like this could become a source of confusion and potential issues if those pages becomes highly trafficked (the Publishing API isn't always the quickest thing to respond to requests asking for content).

I wonder if we can dig a little more in the reasons why we can't get this information from the Search API and whether it's feasible to make changes there?

Sorry this is a bit of a spanner in the works! But I think it's good to make sure we get this right.

Reply:
I agree. I'll clean up git first then investigate.